### DISCUSSION TITLE
How do i download stream file with filename by ByteStream

### DISCUSSION THREAD

**Discussion by Jrohy** [rwf2/Rocket#1741] [Questions] [answered]
i use ByteStream to download sftp file([ssh2](https://docs.rs/ssh2) framework)， but i can't special filename to download file, how to specify filename, please help me(i can't find other framework to download stream file😭)。 this is my example:
```
async fn file_download() -> ByteStream![Vec<u8>]{

    let tcp = TcpStream::connect("1.2.3.4:22").unwrap();
    let mut sess = Session::new().unwrap();
    sess.set_tcp_stream(tcp);
    sess.handshake().unwrap();
    sess.userauth_password("root", "abc").unwrap();

    let file_path = "/root/install-prometheus";
    let (mut remote_file, _stat) = sess.scp_recv(Path::new(file_path)).unwrap();
    ByteStream! {
        let mut buffer = [0; 1000];
        loop {
            if !remote_file.eof() {
                if let Ok(_n) = remote_file.read(&mut buffer[..]) {
                    let data = Vec::from(buffer);
                    yield data;
                } else {
                    break
                }
            } else {
                break
            }
        }
    }
}

```

**Comment by SergioBenitez** [comment, marked as answer]
For the browser to pop up a dialog with a filename, you need to set the `Content-Disposition` header in the response. See [MDN's docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition). It looks like this:

```
Content-Disposition: attachment; filename="filename.jpg"
```

See the [custom responder's section of the guide](https://rocket.rs/v0.5-rc/guide/responses/#custom-responders) for how to easily create a responder with this header.

P.S: Your implementation is incorrect as it makes use of blocking I/O. The majority of your code would be unnecessary if the stream was asynchronous, which it must be to be used in an async context without spawn_blocking. If it was, you could use ReaderStream::one and remove the inefficient buffering in your implementation.

**Comment by Jrohy** [comment]
i try it, thank u @SergioBenitez
